File: simulation/src/objects/Weapon.java
package simulation.src.objects;
import java.util.Random;

public class Weapon {
    private Boolean fist;
    private int health;
    private int dammage;

    public Weapon(Boolean hasWeapon){
        if (hasWeapon){
            Random rand = new Random();
            int rand1 = rand.nextInt(6,10);
            int rand2 = rand.nextInt(20,30);
            health = rand1;
            dammage = rand2;
            fist = false;
        }
        else{
            health = 0;
            dammage = 0;
            fist = true;
        }

    }

    public int getHealth(){
        return health;
        
    }
    public int getDammage(){
        return dammage;
    }
    public boolean getFist(){
        return fist;
    }


}

File: src/Weapon.py
import random

class Weapon:
    def __init__(self, name="fist", health=150, damage=5):
        self.name = name
        self.health = health
        self.damage = damage
        self.fist = name == "fist"

    def get_name(self):
        return self.name

    def get_health(self):
        return self.health

    def get_damage(self):
        return self.damage

    def get_fist(self):
        return self.fist

    def take_damage(self, amount):
        self.health = max(0, self.health - amount)
        return self.health <= 0  # Return True if weapon is broken
